### PR TITLE
Avoid resetting the secret provider on existing stacks

### LINKED ISSUE #197: Documentation in the Stack custom resource definition states to set PULUMI_ACCESS_KEY instead of PULUMI_ACCESS_TOKEN
The documentation in the [Stack custom resource definition](https://github.com/pulumi/pulumi-kubernetes-operator/blob/master/deploy/crds/pulumi.com_stacks.yaml#L36) says to set `PULUMI_ACCESS_KEY` instead of `PULUMI_ACCESS_TOKEN`. Doing so causes an error in the pulumi-kubernetes-operator stating the `PULUMI_ACCESS_TOKEN` is required.

## Steps to reproduce
1. Use EnvRefs with a secret to define the `PULUMI_ACCESS_KEY` in a Stack resource definition.
2. Apply the stack resource.
3. Observe the error logs for the pulumi-kubernetes-operator to see the error `PULUMI_ACCESS_TOKEN must be set for login during non-interactive CLI sessions`.

I believe that the documentation may mean to suggest setting the `PULUMI_ACCESS_TOKEN` instead `PULUMI_ACCESS_KEY`, as setting the `PULUMI_ACCESS_TOKEN` directly solves the issue.

### LINKED ISSUE #201: Secrets provider from stack config getting lost during up
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

As discussed in Slack, when a stack exists and a custom secrets provider is set in the stack config updates fail with:

```
{"level":"debug","ts":1632772421.0293512,"logger":"controller_stack","msg":"stackConfig loaded","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","stack":{},"stackConfig":{"secretsprovider":"awskms://alias/REDACTED","encryptedk
ey":"REDACTED","config":{"tetrat
e:metadata":{"owner":"maintainability","source":"cloud/projects/operations/storage/aws"}}}}
{"level":"debug","ts":1632772421.770783,"logger":"controller_stack","msg":"Updated stack config","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","Stack.Name":"operations.storage.aws","config":{}}
{"level":"debug","ts":1632772421.771033,"logger":"controller_stack","msg":"InstallProjectDependencies","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","workspace":"/tmp/pulumi_auto636600156/cloud/projects/operations/storage/aws"}
{"level":"debug","ts":1632772422.9292898,"logger":"controller_stack","msg":"NPM/Yarn","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","Dir":"/tmp/pulumi_auto636600156/cloud/projects/operations/storage/aws","Path":"/usr/bin/yarn","Args":["/usr
/bin/yarn","install"],"Stdout":"➤ YN0000: ┌ Resolution step"}
...
{"level":"debug","ts":1632772426.5784225,"logger":"controller_stack","msg":"NPM/Yarn","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","Dir":"/tmp/pulumi_auto636600156/cloud/projects/operations/storage/aws","Path":"/usr/bin/yarn","Args":["/usr
/bin/yarn","install"],"Stdout":"➤ YN0000: Done with warnings in 3s 651ms"}
{"level":"info","ts":1632772426.6382632,"logger":"controller_stack","msg":"Checking current HEAD commit hash","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","Current commit":"REDACTED"}
{"level":"info","ts":1632772426.6382985,"logger":"controller_stack","msg":"New commit hash found","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","Current commit":"REDACTED","Last commit":""}
{"level":"error","ts":1632772428.426037,"logger":"controller_stack","msg":"Failed to refresh stack","Request.Namespace":"pulumi-operator","Request.Name":"operations.storage.aws","Stack.Name":"operations.storage.aws","error":"refreshing stack \"operations.storage.aws\": fa
iled to refresh stack: exit status 255\ncode: 255\nstdout: \nstderr: warning: A new version of Pulumi is available. To upgrade from version '3.11.0' to '3.13.0', visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.\nerror: getting se
crets manager: passphrase must be set with PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE environment variables\n\n","errorVerbose":"failed to refresh stack: exit status 255\ncode: 255\nstdout: \nstderr: warning: A new version of Pulumi is available. To upgrade
 from version '3.11.0' to '3.13.0', visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.\nerror: getting secrets manager: passphrase must be set with PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE environment variables\n\n
\nrefreshing stack \"operations.storage.aws\"\ngithub.com/pulumi/pulumi-kubernetes-operator/pkg/controller/stack.(*reconcileStackSession).RefreshStack\n\t/home/runner/work/pulumi-kubernetes-operator/pulumi-kubernetes-operator/pkg/controller/stack/stack_controller.go:814\n
github.com/pulumi/pulumi-kubernetes-operator/pkg/controller/stack.(*ReconcileStack).Reconcile\n\t/home/runner/work/pulumi-kubernetes-operator/pulumi-kubernetes-operator/pkg/controller/stack/stack_controller.go:237\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*
Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/control
ler-runtime@v0.9.0/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:214\nruntime.goexit\n\t/
opt/hostedtoolcache/go/1.16.2/x64/src/runtime/asm_amd64.s:1371","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:298\n
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.
func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:214"}
```

I was able to work around this by adding the secret provider in the stack CR, but this shouldn't be required as it's already in the stack yaml file.

### Steps to reproduce

Stack config:

```
spec:
  backend: s3://REDACTED
  branch: refs/heads/master
  gitAuth:
    sshAuth:
      sshPrivateKey:
        type: Secret
        ...
  projectRepo: ...
  refresh: true
  repoDir: cloud/projects/operations/storage/aws
  stack: operations.storage.aws
  useLocalStackOnly: true
```

Expected: Update to succeed.
Actual: See error ^^

### PR DESCRIPTION
**Pull request by viveklak**
Also adds a test for S3 backed backend and KMS key based secret encryption.

Fixes #201
Fixes #197

### CODE REVIEW

File: .github/workflows/run-acceptance-tests.yaml [hunk 1]
```diff
@@ -85,7 +85,9 @@ jobs:
         run: |
           # Create GKE test cluster to install CRDs and use with the test operator.
           scripts/ci-cluster-create.sh
-
+          # Source the env variables created in the script above
+          cat ~/.envfile
```
Comment (lblackstone) [inline .github/workflows/run-acceptance-tests.yaml:89]: Is there anything sensitive in the env file that we don't want appearing in CI logs?

File: pkg/apis/pulumi/v1alpha1/stack_types.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ type StackSpec struct {
 	SecretRefs map[string]ResourceRef `json:"secretsRef,omitempty"`
 	// (optional) SecretsProvider is used to initialize a Stack with alternative encryption.
 	// Examples:
-	//   - AWS:   "awskms://arn:aws:kms:us-east-1:111122223333:key/1234abcd-12ab-34bc-56ef-1234567890ab?region=us-east-1"
+	//   - AWS:   "awskms:///arn:aws:kms:us-east-1:111122223333:key/1234abcd-12ab-34bc-56ef-1234567890ab?region=us-east-1"
 	//   - Azure: "azurekeyvault://acmecorpvault.vault.azure.net/keys/mykeyname"
 	//   - GCP:   "gcpkms://projects/MYPROJECT/locations/MYLOCATION/keyRings/MYKEYRING/cryptoKeys/MYKEY"
```
Comment (lblackstone) [inline pkg/apis/pulumi/v1alpha1/stack_types.go:62]: Are the URI formats for Azure and GCP correct?

**Review by lblackstone — APPROVED** [review]

File: .github/workflows/run-acceptance-tests.yaml (hunk 1 above)
Comment (viveklak) [reply .github/workflows/run-acceptance-tests.yaml:89]: Just the backend s3 bucket and the KMS key ARN. Neither are sensitive.

File: pkg/apis/pulumi/v1alpha1/stack_types.go (hunk 2 above)
Comment (viveklak) [reply pkg/apis/pulumi/v1alpha1/stack_types.go:62]: Yes - at least based on what I see here: https://www.pulumi.com/docs/intro/concepts/secrets/#initializing-a-stack-with-alternative-encryption

**Pull request merged**